extensions/notebook/src/test/book/book.test.ts

```diff
@@ -218,15 +224,65 @@ describe('BookTreeViewProviderTests', function () {
 
 			});
 
-			it('providedBookTreeViewProvider should return all book nodes when element is undefined', async function (): Promise<void> {
+			it('getParent should return when element is a valid child notebook', async () => {
+				let parent = await bookTreeViewProvider.getParent();
+				should(parent).be.undefined();
+
+				parent = await bookTreeViewProvider.getParent(notebook2);
+				should(parent).not.be.undefined();
+				equalBookItems(parent, expectedNotebook1);
+			});
+
+			it('revealActiveDocumentInViewlet should return correct bookItem for highlight', async () => {
+				let notebook1Path = path.join(rootFolderPath, 'Book', 'content', 'notebook1.ipynb');
+				let currentSelection = await bookTreeViewProvider.findAndExpandParentNode(notebook1Path);
+				equalBookItems(currentSelection, expectedNotebook1);
+			});
+
+			it('revealActiveDocumentInViewlet should be called on showNotebook', async () => {
+				let notebook1Path = path.join(rootFolderPath, 'Book', 'content', 'notebook1.ipynb');
+				let notebook2Path = path.join(rootFolderPath, 'Book', 'content', 'notebook2.ipynb');
+				let notebookUri = vscode.Uri.file(notebook2Path);
+
+				let revealActiveDocumentInViewletSpy = sinon.spy(bookTreeViewProvider, 'revealActiveDocumentInViewlet');
+				await azdata.nb.showNotebookDocument(notebookUri);
+				should(azdata.nb.notebookDocuments.find(doc => doc.fileName === notebookUri.fsPath)).not.be.undefined();
+				should(revealActiveDocumentInViewletSpy.calledOnce).be.true('revealActiveDocumentInViewlet should have been called');
+
+				notebookUri = vscode.Uri.file(notebook1Path);
+				await azdata.nb.showNotebookDocument(notebookUri);
+				should(azdata.nb.notebookDocuments.find(doc => doc.fileName === notebookUri.fsPath)).not.be.undefined();
+				should(revealActiveDocumentInViewletSpy.calledTwice).be.true('revealActiveDocumentInViewlet should have been called twice');
+				sinon.restore();
```

Comment:
Just use an afterEach hook at the top level describe to run sinon.restore after each test

Reply:
I've used sinon in few tests only, so didn't make that afterEach hook, I guess if it isn't a lot of overhead I can add that.